patternfly/patternfly-elements
# Fix/a11y update pfe clipboard

## Updated accessibility features for pfe-clipboard

- Address some issues found by Lighthouse and aXe DevTools
- Fixed some comment typos
- Added appropriate aria features and js updates
- Addressed major issues found during screen reader testing.
- Updated the component to function for all users rather than just sighted mouse users
- Added default state of 4 seconds and logic to ensure that the timeout is never set below 4 seconds so that users have enough time to hear and read the text state changes

### Related issues

#1873


### Preview

- Check out the fix/a11y-update-pfe-clipboard branch
- Run npm start
- Go to http://localhost:8000/demo/pfe-clipboard


### What has changed and why



- `pfe-clipboard.ts`
   - Added new state property for aria-disabled to added aria features
   - Added comments for changes
   - Updated the HTML in render()  to add aria features
   - Cleaned up some comment typos
- `pfe-clipboard.scss`
   - Added sr-only class to utilize with pfe-clipboard to improve the success message output for screen readers
   - Adjusted the padding and changes some ids to be classes to go with new HTML structure
- `pfe-clipboard.html`
   - Removed role button and tabindex from pfe-clipboard tags because that is being set within the shadowDom now
- `pfe-clipboard.js`
   - Removed role button and tabindex from pfe-clipboard tags because that is being set within the shadowDom now

### Testing instructions



  1. Follow the link to the demo
  2. Turn on the screen reader that goes with your OS
  3. Tab through the demo section for pfe-clipboard and activate the buttons
     - Expected behavior: the visual text changes to "Copied" and the screen reader reads out loud the success message. Then after at least 4 seconds, the copy feature flips back to the initial state.

#### Browser requirements

Your component should work in all of the following environments:



### Ready-for-merge Checklist




### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
